src/meetingreadinesschecker/DefaultMeetingReadinessChecker.ts

```diff
@@ -0,0 +1,451 @@
+      return CheckVideoInputFeedback.Failure;
+    }
+  }
+
+  async checkCameraResolution(
+    videoInputDevice: MediaDeviceInfo,
+    width: number,
+    height: number
+  ): Promise<CheckCameraResolutionFeedback> {
+    const videoConstraint = {
+      video: { width: { exact: width }, height: { exact: height } },
+    };
+    let stream: MediaStream;
+    try {
+      stream = await navigator.mediaDevices.getUserMedia(videoConstraint);
+    } catch (error) {
+      this.logger.error(
+        `MeetingReadinessChecker: Camera resolution check with width: ${width} height ${height} failed with error ${error}`
+      );
+      if (error && error.name === 'OverconstrainedError') {
+        return CheckCameraResolutionFeedback.ResolutionNotSupported;
+      }
+      if (error && error.name === 'NotAllowedError') {
+        return CheckCameraResolutionFeedback.PermissionDenied;
+      }
+      return CheckCameraResolutionFeedback.Failure;
+    } finally {
+      if (stream) {
+        stream.getTracks().forEach(function(track) {
+          track.stop();
+        });
+      }
+    }
+    return CheckCameraResolutionFeedback.Success;
+  }
+
+  async checkContentShareConnectivity(): Promise<CheckContentShareConnectivityFeedback> {
+    let isContentShareStarted = false;
+    let isAudioVideoStarted = false;
+
+    const contentShareObserver: ContentShareObserver = {
+      contentShareDidStart: () => {
+        isContentShareStarted = true;
+      },
+    };
+    const observer: AudioVideoObserver = {
+      audioVideoDidStart: () => {
+        isAudioVideoStarted = true;
+      },
+    };
+
+    try {
+      this.meetingSession.audioVideo.addObserver(observer);
+      this.meetingSession.audioVideo.start();
+
+      this.meetingSession.audioVideo.addContentShareObserver(contentShareObserver);
+      await this.meetingSession.audioVideo.startContentShareFromScreenCapture();
+
+      await this.executeTimeoutTask(async () => {
+        return isAudioVideoStarted && isContentShareStarted;
+      });
+
+      this.meetingSession.audioVideo.removeObserver(observer);
+      if (!isAudioVideoStarted) {
+        return CheckContentShareConnectivityFeedback.ConnectionFailure;
+      }
+      await this.stopMeeting();
+      return isContentShareStarted
+        ? CheckContentShareConnectivityFeedback.Success
+        : CheckContentShareConnectivityFeedback.Timeout;
+    } catch (error) {
+      this.logger.error(`MeetingReadinessChecker: Content share check failed with error ${error}`);
+      if (error.name === 'NotAllowedError') {
+        return CheckContentShareConnectivityFeedback.PermissionDenied;
+      } else {
+        return CheckContentShareConnectivityFeedback.Failure;
+      }
+    } finally {
+      this.meetingSession.audioVideo.stopContentShare();
+      this.meetingSession.audioVideo.removeContentShareObserver(contentShareObserver);
+    }
+  }
+
+  async checkAudioConnectivity(
```

Comment:
(Don't need to change in this PR) I think I shared before that you could set this value to a non-zero value and just check whether the meeting gets started.
https://github.com/aws/amazon-chime-sdk-js/blob/master/src/meetingsession/MeetingSessionConfiguration.ts#L53

Reply:
I did not know about this. it indeed makes the API simpler.

Reply:
I thought about it a bit more and feel like if we do it like that then we could not differentiate between audioVideoDidStart fail and no attendee presence. If it does not matter to differentiate between those then I group it up but we need to do it now since I dont think we can change the enum later.